src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -402,6 +402,12 @@ private static bool NeedsSeparator(SyntaxToken token, SyntaxToken next)
                 return true;
             }
 
+            // We don't want to add extra space after cast, we want space only after tuple
+            if (token.IsKind(SyntaxKind.CloseParenToken) && IsWord(next.Kind()) && token.Parent?.IsKind(SyntaxKind.TupleType) == true)
```

Comment:
1. i don't understand what the IsWord check is for.  
2. you don't need `?.` after token.Parent.  All tokens you could reach in SyntaxNormalizer have parents.
3. I'd much rather we enumerate the cases where a tuple type would want a space after it (or where it woudl not want a space after it).  Looking specifically for `isword` just seems busted.

Reply:
In cases like:
```csharp
(string a, (string b, string c))
```
I don't want to add space after `c)` so I want to have space only after `)<identifier,word>` the same expression exist above but for braket: https://github.com/dotnet/roslyn/blob/ec99ff789c32a34fe444f8631e5a5166d2ebc11d/src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs#L400

Reply:
So I think we want space after tuple only when next token is word.